tests/test_elements/test_mol2mol/test_mol2scaffold.py

```diff
@@ -0,0 +1,104 @@
+"""Test the mol2scaffold module."""
+
+from typing import Any
+from unittest import TestCase
+
+from molpipeline import Pipeline
+from molpipeline.any2mol import AutoToMol
+from molpipeline.mol2any import MolToSmiles
+from molpipeline.mol2mol.scaffolds import MakeScaffoldGeneric, MurckoScaffold
+
+
+class TestMurckoScaffold(TestCase):
+    """Test the MurckoScaffold class."""
+
+    def test_murcko_scaffold_generation_pipeline(self) -> None:
+        """Test the scaffold generation."""
+        scaffold_pipeline = Pipeline(
+            steps=[
+                ("smiles_to_mol", AutoToMol()),
+                ("murcko_scaffold", MurckoScaffold()),
+                ("scaffold_to_smiles", MolToSmiles()),
+            ]
+        )
+        smiles_list = ["Cc1ccc(=O)[nH]c1", "O=CC1CCC(c2ccccc2)CC1", "CCC"]
+        expected_scaffold_list = ["O=c1cccc[nH]1", "c1ccc(C2CCCCC2)cc1", ""]
+
+        scaffold_list = scaffold_pipeline.transform(smiles_list)
+        for scaffold, expected_scaffold in zip(scaffold_list, expected_scaffold_list):
```

Comment:
The default behavior of `zip` does not check for the same length of the two iterables but stops the iteration if one is exhausted. You should test the length before the loop, like `self.assertEqual(len(scaffold_list), len(expected_scaffold_list))`. Alternatively, you could use the `strict` argument of `zip` (see https://stackoverflow.com/a/62773251)

Reply:
switched to self.assertListEqual.